### ISSUE TITLE
Write Docs for the Project

### ISSUE THREAD

**Issue by pl-aknight** [q2ebanking/boa-constrictor#6]
The README and TUTORIAL are great, but we should invest in more comprehensive documentation for the project. GitHub wikis would be great.

Topics to cover:

- The Screenplay Pattern
- Using Boa Constrictor
- Available Questions and Tasks
- Project information

**Comment by SabotageAndi** [comment]
My two cents about GitHub Wiki as we used it in the past in SpecFlow: They look nice at the first glance, but are limiting very quickly.
E.g. you can't make a PR against it. So you have only the option to make it editable by everyone and you will not notice any changes or only members of the repository can edit it, which means lesser contributions.

Because of this, we at SpecFlow moved to ReadTheDocs. Now we have the docs in the same repo as the code and we can update them at the same time with the same PR. So code and documentation don't apart away from each other.

**Comment by pl-aknight** [comment]
Thanks @SabotageAndi! Great insights.

**Comment by pl-aknight** [comment]
https://www.youtube.com/watch?v=t4vKPhjcMZg

**Comment by AndyLPK247** [comment]
https://www.youtube.com/watch?v=4SwdVMKhbn4

**Comment by AndyLPK247** [comment]
Options:

- https://www.mkdocs.org/
- https://www.sphinx-doc.org/en/master/
- https://docs.readthedocs.io/en/stable/index.html

**Comment by AndyLPK247** [comment]
I think I favor using MkDocs and publishing the docs to ReadTheDocs.

**Comment by AndyLPK247** [comment]
Apparently, GitHub Pages recommends Jekyll:
https://jekyllrb.com/docs/

**Comment by AndyLPK247** [comment]
GitHub Pages + Jekyll seems to be the simplest way to go. We can keep everything within GitHub.

**Comment by pl-aknight** [comment]
It looks like the best way to create a `docs` folder and make it appear in Visual Studio is to create the folder in the repository directory, add it as an "Existing Web Site" to the .NET Solution in VS, and then remove it from the solution's build. Steps are documented here: https://stackoverflow.com/questions/267200/visual-studio-solutions-folder-as-real-folders. I tried it, and it works decently well.

**Comment by pl-aknight** [comment]
We also face a bit of a dilemma with the GitHub Pages + Jekyll. There are two approaches to writing docs with these tools:

1. Lean - Put Markdown files and static assets into the `/docs` folder, and GitHub will publish them upon merge into `master`.
2. Hands on - Set up a full Jekyll site for local development and testing, then push to GitHub.

I verified that approach 1 works. It would be easiest for letting contributors edit the docs. However, we can't view pages before they are published. That's not good for large changes.

I tried approach 2 locally, and I could get my Jekyll project running locally. I had to install a bunch of Ruby stuff, but it looks like we don't need to do any Ruby programming. The template website Jekyll creates also shows how to do some advanced templating, which could be great. However, this setup is much more intense than simply dropping `.md` files into a `/docs' folder. I think approach 1 can work for small changes, but serious doc editors should use approach 2. (I wish there were an easier way to test pages before pushing them live.)
